packages/flutter/test/material/popup_menu_test.dart

```diff
@@ -851,6 +851,139 @@ void main() {
+    // [PopupMenuButton] should assert that either child == null, icon == null,
+    // or that both are null because only one of the two can be displayed/used.
+    expect(() => PopupMenuButton<int>(
+      itemBuilder: (BuildContext context) => <PopupMenuItem<int>>[],
+      child: Container(),
+      icon: const Icon(Icons.error),
+    ), throwsAssertionError);
+  });
+
+  testWidgets('PopupMenuButton default tooltip', (WidgetTester tester) async {
+    // If the tooltip parameter for [PopupMenuButton] is not specified,
+    // a default tooltip should be used instead.
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Material(
+          child: Column(
+            // The default tooltip should be present when
+            // [PopupMenuButton.icon] or [PopupMenuButton.child] is defined
+            // and also when neither of the two is defined.
+            children: <Widget>[
+              // Default Tooltip should be present when [PopupMenuButton.child]
+              // and [PopupMenuButton.child] are undefined.
+              PopupMenuButton<int>(
+                itemBuilder: (BuildContext context) {
+                  return <PopupMenuEntry<int>>[
+                    const PopupMenuItem<int>(
+                      value: 1,
+                      child: Text('Tap me please!'),
+                    ),
+                  ];
+                },
+              ),
+              // Default Tooltip should be present when
+              // [PopupMenuButton.child] is defined.
+              PopupMenuButton<int>(
+                itemBuilder: (BuildContext context) {
+                  return <PopupMenuEntry<int>>[
+                    const PopupMenuItem<int>(
+                      value: 1,
+                      child: Text('Tap me please!'),
+                    ),
+                  ];
+                },
+                child: const Text('Test text'),
+              ),
+              // Default Tooltip should be present when
+              // [PopupMenuButton.icon] is defined.
+              PopupMenuButton<int>(
+                itemBuilder: (BuildContext context) {
+                  return <PopupMenuEntry<int>>[
+                    const PopupMenuItem<int>(
+                      value: 1,
+                      child: Text('Tap me please!'),
+                    ),
+                  ];
+                },
+                icon: const Icon(Icons.check),
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+
+    // The default tooltip is defined as [MaterialLocalizations.showMenuTooltip]
+    // and it is used when no tooltip is provided.
+    expect(find.byType(Tooltip), findsNWidgets(3));
+    expect(find.byTooltip(const DefaultMaterialLocalizations().showMenuTooltip), findsNWidgets(3));
+  });
+
+  testWidgets('PopupMenuButton custom tooltip', (WidgetTester tester) async {
+    // A custom tooltip can be provided to [PopupMenuButton.tooltip]
```

Comment:
Also, this comment seems redundant, since the test name implies that we're testing for custom tooltips